hazelcast/src/main/java/com/hazelcast/spi/serialization/SerializationService.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.serialization;
+
+import com.hazelcast.core.ManagedContext;
+import com.hazelcast.core.PartitioningStrategy;
+import com.hazelcast.nio.serialization.Data;
+
+/**
+ * SPI to serialize user objects to {@link Data} and back to Object
+ * {@link Data} is the internal representation of binary data in hazelcast.
+ */
+public interface SerializationService {
```

Comment:
Can you add a line of comment that this class can't be implemented by an end user since we want to have the freedom to add new methods.